### PR TITLE
MINIFICPP-394: Implement MQTT C2 protocol along with MQTT/REST

### PR DESCRIPTION
**Pull request by phrocker**
translation layer

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by phrocker** [comment]
I noticed there were no MQTT tests. Before this is merged I want to change this in some way. Plan was to use a stubbed out MQTT controller service to test the business logic.

File: extensions/mqtt/controllerservice/MQTTControllerService.h [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+        ssl_context_service_(nullptr),
+        logger_(logging::LoggerFactory<MQTTControllerService>::getLogger()) {
+    setConfiguration(configuration);
+    initialize();
+  }
+
+  static core::Property BrokerURL;
+  static core::Property ClientID;
+  static core::Property UserName;
+  static core::Property PassWord;
```
[the remark this answers is not in the document]
Comment (phrocker) [reply extensions/mqtt/controllerservice/MQTTControllerService.h]: Not nitpicky at all. I moved code that already existed, so this was already there. Agree with you. Will fix.

**Comment by phrocker** [comment]
@apiri  https://github.com/apache/nifi-minifi-cpp/pull/324 should solve the build failure. I see I have conflicts so I'll rebase too

**Comment by apiri** [comment]
Was able to verify #321 and rebase this on that.  Build looks good here.

**Pull request closed without merging**